plugins/funind/recdef.ml

```diff
@@ -1528,14 +1528,14 @@ let recursive_definition is_mes function_name rec_impls type_of_f r rec_arg_num
   let open Constr in
   let open CVars in
   let env = Global.env() in
-  let evd = ref (Evd.from_env env) in
-  let function_type = interp_type_evars env evd type_of_f in
+  let evd = Evd.from_env env in
+  let evd, function_type = interp_type_evars env evd type_of_f in
   let function_type = EConstr.Unsafe.to_constr function_type in
   let env = push_named (Context.Named.Declaration.LocalAssum (function_name,function_type)) env in
   (* Pp.msgnl (str "function type := " ++ Printer.pr_lconstr function_type);  *)
-  let ty = interp_type_evars env evd ~impls:rec_impls eq in
+  let evd, ty = interp_type_evars env evd ~impls:rec_impls eq in
   let ty = EConstr.Unsafe.to_constr ty in
-  let evm, nf = Evarutil.nf_evars_and_universes !evd in
+  let eve, nf = Evarutil.nf_evars_and_universes evd in
```

Comment:
There is a typo here I think, that should probably be `evm` instead of `eve`.

Reply:
Yup, fixed thanks! I tend to run with unused variables warning on, but it seems in this file I forgot.

Reply:
We should enable that warning by default, really.

Reply:
Writing a PR now.

Reply:
Unused variable doesn't warn for patterns where at least 1 variable is used, and I guess here `nf` is used.
Innocuous unused would warn but enabling it is too painful.

Reply:
See https://github.com/SkySkimmer/coq/commits/messing-around where I tried enabling innocuours unused but abandoned on the way.

Reply:
I am almost done with a new try.

Reply:
It is painful? A bit. It is worth it? IMHO totally.

Reply:
Specially in the upper layers.